### ISSUE TITLE
turn on "require PRS up to date" branch control

### ISSUE THREAD

**Issue by sneakers-the-rat** [pyOpenSci/pyos-package-template#64]
What do ppl think about adding the "require branches be up to date before merging" branch protection rule to main? I think it helps avoid problems when there are lots of overlapping changes happening, that e.g. a merge that passed tests on its last commit could still break things if the base has changed since then?

**Comment by blink1073** [comment]
Looks like we have quorum, I've added the rule.

<img width="666" alt="image" src="https://github.com/user-attachments/assets/8dca971e-583b-443b-833b-c61ab7d33bb7">

**Comment by sneakers-the-rat** [comment]
I don't mean to be annoying but it looks like that's a different rule? this was the one i was thinking of:

<img width="836" alt="Screenshot 2024-10-31 at 4 35 08 PM" src="https://github.com/user-attachments/assets/81ac3688-489d-4e14-8f9d-8859045083da">

I am also fine with requiring pull request before merging, and have found 'dismiss stale approvals' to be *slightly* annoying in combination with 'require branches be up to date' because it was pretty common with us to review something, approve it, then before merging do a final rebase and that triggers the need for a new review. maybe since this isn't a "security-conscious" package and we're not anticipating people trying to sneak through changes we disable that? but also no strong preference

**Comment by Midnighter** [comment]
I agree, especially with a globally distributed group of volunteers, let's not be too strict.

**Comment by blink1073** [comment]
Updated!

**Comment by lwasser** [comment]
hi!! I'm late to the convo here, but I totally trust all the decisions that you all make on this. so we could potentially close this issue or keep it opened as you all see fit!! i don't want to slow progress down!
